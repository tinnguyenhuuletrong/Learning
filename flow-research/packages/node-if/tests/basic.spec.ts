import NodeMeta from "../src/index";

describe("basic", () => {
  it("create and run", async () => {
    const ins = NodeMeta.create({
      condition: {
        age: { $gt: 18 },
      },
      trueNode: "Upper 18",
      falseNode: "Under 18",
    });
    let res;
    res = await ins.exec({ age: 10 });
    expect(res).toMatchSnapshot();

    res = await ins.exec({ age: 20 });
    expect(res).toMatchSnapshot();
  });
});
