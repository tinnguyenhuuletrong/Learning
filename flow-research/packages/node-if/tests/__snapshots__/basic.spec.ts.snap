// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic create and run 1`] = `
Object {
  "isSuccess": true,
  "nextNode": "Under 18",
}
`;

exports[`basic create and run 2`] = `
Object {
  "isSuccess": true,
  "nextNode": "Upper 18",
}
`;
